components/entity-access-rules/api/src/main/java/org/phenotips/data/permissions/PermissionsManager.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+package org.phenotips.data.permissions;
+
+import org.phenotips.data.Patient;
+import org.phenotips.entities.PrimaryEntity;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.stability.Unstable;
+
+import java.util.Collection;
+import java.util.Iterator;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * @version $Id$
+ * @since 1.0M9
+ */
+@Unstable
+@Role
+public interface PermissionsManager
+{
+    /**
+     * Get the visibility options available, excluding {@link Visibility#isDisabled() disabled} ones.
+     *
+     * @return a collection of enabled {@link Visibility} visibilities; may be empty if none are enabled
+     */
+    @Nonnull
+    Collection<Visibility> listVisibilityOptions();
+
+    /**
+     * Get all visibility options available in the platform, including {@link Visibility#isDisabled() disabled} ones.
+     *
+     * @return a collection of {@link Visibility} visibilities; may be empty if none are available
+     */
+    @Nonnull
+    Collection<Visibility> listAllVisibilityOptions();
+
+    /**
+     * Get the default {@link Visibility} visibility to set for new patient records.
+     *
+     * @return a configured {@link Visibility visibility} if valid, or default
+     *         {@link org.phenotips.data.permissions.internal.visibility.PrivateVisibility}
+     */
+    @Nonnull
+    Visibility getDefaultVisibility();
+
+    /**
+     * Get the {@link Visibility} visibility from its {@code name}. Defaults to
+     * {@link org.phenotips.data.permissions.internal.visibility.PrivateVisibility} if {@code name} is invalid.
+     *
+     * @param name the desired visibility name, as string
+     * @return the {@link Visibility} associated with the provided {@code name}, will default to
+     *         {@link org.phenotips.data.permissions.internal.visibility.PrivateVisibility} if {@code name} is invalid
+     */
+    @Nonnull
+    Visibility resolveVisibility(@Nullable String name);
+
+    @Nonnull
+    Collection<AccessLevel> listAccessLevels();
+
+    @Nullable
+    AccessLevel resolveAccessLevel(@Nullable String name);
```

Comment:
What i sit supposed to mean mean when a `null` is returned - is it "no access"?

Reply:
OK, never mind, I found a comment in another class, probably should add it here:

`@return the {@link AccessLevel} associated with the provided {@code name}, or null if the provided {@code name}`
 +     * is not valid